operator/controllers/coreprovider_controller.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1alpha4 "sigs.k8s.io/cluster-api/operator/api/v1alpha4"
+)
+
+// CoreProviderReconciler reconciles a CoreProvider object
+type CoreProviderReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
```

Comment:
Is this required?
(same for the other providers)

Reply:
It is used in `operator/main.go`, when the reconciler instance is created.